### DISCUSSION TITLE
Removal listener called multiple times for the same object.

### DISCUSSION THREAD

**Discussion by valmaltsev** [ben-manes/caffeine#679] [Q&A] [answered]
I noticed that when cache is configured with both maximumSize() and expireAfterAccess(), sometimes removal listener called twice for the same object, once with SIZE and once with EXPIRED causes. Is this by design?

**Comment by ben-manes** [comment, marked as answer]
It should not. Can you provide a unit test that demonstrates the problem? There were some minor bugs fixed in 3.0.4 (https://github.com/ben-manes/caffeine/issues/593) if you are on an older release.

**Comment by valmaltsev** [comment]
We're on 3.0.5. I don't have a unit test right now. It was observed in performance tests of our product. I'll try to reproduce it in a unit test.

**Comment by valmaltsev** [comment]
I can't come up with a unit test that reproduces the issue. I guess it was caused by a bug in our own code.

**Reply by ben-manes** [reply]
Thanks for trying!

**Comment by ben-manes** [comment]
I've been mining the code coverage report for ideas on new test cases and I think that I discovered 3 bugs, two of which are for removal notifications. They don't match with your scenario, though. I'm still iterating through the CI to squash mistakes, as it takes ~2hrs per build due to test volume. You can see this [commit](https://github.com/ben-manes/caffeine/commit/59160d961e930d1022be6ecab053cc0e60f1ddf5) for an in-progress iteration and the bug descriptions.
